src/js/tools/sentryMetroSerializer.ts

```diff
@@ -0,0 +1,167 @@
+import * as crypto from 'crypto';
+import type { MixedOutput, Module } from 'metro';
+import CountingSet from 'metro/src/lib/CountingSet';
+import * as countLines from 'metro/src/lib/countLines';
+import { createDefaultMetroSerializer } from './vendor/metro/utils';
+import { Bundle, MetroSerializer, determineDebugIdFromBundleSource, stringToUUID, VirtualJSOutput, MetroSerializerOutput, SerializedBundle, createDebugIdSnippet } from './utils';
+
+type SourceMap = Record<string, unknown>;
+
+const DEBUG_ID_PLACE_HOLDER = '__debug_id_place_holder__';
+const DEBUG_ID_MODULE_PATH = '__debugid__';
+const PRELUDE_MODULE_PATH = '__prelude__';
+const SOURCE_MAP_COMMENT = '//# sourceMappingURL=';
+const DEBUG_ID_COMMENT = '//# debugId=';
+
+/**
+ * Creates a Metro serializer that adds Debug ID module to the plain bundle.
+ * The Debug ID module is a virtual module that provides a debug ID in runtime.
+ *
+ * RAM Bundles do not support custom serializers.
+ */
+export const createSentryMetroSerializer = (customSerializer?: MetroSerializer): MetroSerializer => {
+  return async function (entryPoint, preModules, graph, options) {
+    const serializer = customSerializer || createDefaultMetroSerializer();
```

Comment:
l: let's move this call above the `return async function` so we don't construct default serializer every time?